src/main/java/bio/terra/workspace/service/iam/SamService.java

```diff
@@ -72,4 +78,24 @@ public void workspaceAuthz(AuthenticatedUserRequest userReq, String workspaceId,
               + workspaceId
               + " or it doesn't exist.");
   }
+
+  public SystemStatusSystems status() {
+    // No access token needed since this is an unauthenticated API.
+    StatusApi statusApi = new StatusApi(getApiClient(null));
+
+    try {
+      // Note the SystemStatus class here is from the Sam client library, not generated by WM's
+      // OpenAPI, so this requires some amount of reshaping objects.
+      SystemStatus samStatus = statusApi.getSystemStatus();
+      Map<String, Map<String, String>> subsystemStatusMap =
```

Comment:
I'd like other people's thoughts on this one - samStatus.getSystems() returns an Object because that's all that's guaranteed by their API, but it feels really fragile to be casting an Object like this. 
An alternative is including the `getSystems().toString()` as a single message, but that means we lose machine-readability of Sam subsystems in this message. Thoughts on this tradeoff?

Reply:
hmm. `samStatus.getSystems()` should return a `Map<String, SubsystemStatus>` and the fact that it doesn't is a problem in the Sam client library (which is a result of awkward swagger definitions, and I don't believe is trivial to fix).

I somehow doubt that casting directly to `Map<String, SubsystemStatus>` will work. What if you serialized the output of `samStatus.getSystems()` to JSON, then deserialized from JSON to `Map<String, SubsystemStatus>`, with error-trapping? (is that too heavyweight?)

Reply:
I like that solution better than casting directly, especially since failures give us JSON exceptions instead of null pointer exceptions.